kroxylicious/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java

```diff
@@ -212,22 +217,21 @@ else if ((state == State.START
                     && msg instanceof RequestFrame) {
                 bufferMsgAndSelectServer(msg);
             }
+            else if (state == State.CONNECTING || state == State.CONNECTED) {
```

Comment:
I get that this is consistent with how the other guard conditions are done, but I think I'd like to see them all extracted as methods. e.g. `shouldBufferFrame` or `shouldBufferMessage`

Reply:
I've done a bit of method extraction now if you want to have a look at the naming